x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvSort.java

```diff
@@ -0,0 +1,243 @@
+        boolean ordering = order.foldable() && ((BytesRef) order.fold()).utf8ToString().equalsIgnoreCase("DESC") ? false : true;
+        return switch (PlannerUtils.toElementType(field.dataType())) {
+            case BOOLEAN -> new MvSort.EvaluatorFactory(
+                toEvaluator.apply(field),
+                ordering,
+                (blockFactory, fieldBlock, sortOrder) -> new MultivalueDedupeBoolean((BooleanBlock) fieldBlock).sortToBlock(
+                    blockFactory,
+                    sortOrder
+                )
+            );
+            case BYTES_REF -> new MvSort.EvaluatorFactory(
+                toEvaluator.apply(field),
+                ordering,
+                (blockFactory, fieldBlock, sortOrder) -> new MultivalueDedupeBytesRef((BytesRefBlock) fieldBlock).sortToBlock(
+                    blockFactory,
+                    sortOrder
+                )
+            );
+            case INT -> new MvSort.EvaluatorFactory(
+                toEvaluator.apply(field),
+                ordering,
+                (blockFactory, fieldBlock, sortOrder) -> new MultivalueDedupeInt((IntBlock) fieldBlock).sortToBlock(blockFactory, sortOrder)
+            );
+            case LONG -> new MvSort.EvaluatorFactory(
+                toEvaluator.apply(field),
+                ordering,
+                (blockFactory, fieldBlock, sortOrder) -> new MultivalueDedupeLong((LongBlock) fieldBlock).sortToBlock(
+                    blockFactory,
+                    sortOrder
+                )
+            );
+            case DOUBLE -> new MvSort.EvaluatorFactory(
+                toEvaluator.apply(field),
+                ordering,
+                (blockFactory, fieldBlock, sortOrder) -> new MultivalueDedupeDouble((DoubleBlock) fieldBlock).sortToBlock(
+                    blockFactory,
+                    sortOrder
+                )
+            );
+            case NULL -> EvalOperator.CONSTANT_NULL_FACTORY;
+            default -> throw new IllegalArgumentException("unsupported type [" + field.dataType() + "]");
+        };
+    }
+
+    private record EvaluatorFactory(
+        EvalOperator.ExpressionEvaluator.Factory field,
+        boolean order,
+        TriFunction<BlockFactory, Block, Boolean, Block> sort
+    ) implements EvalOperator.ExpressionEvaluator.Factory {
+        @Override
+        public EvalOperator.ExpressionEvaluator get(DriverContext context) {
+            return new MvSort.Evaluator(context.blockFactory(), field.get(context), order, sort);
+        }
+
+        @Override
+        public String toString() {
+            return "MvSort[field=" + field + ", order=" + order + "]";
+        }
+    }
+
+    private static class Evaluator implements EvalOperator.ExpressionEvaluator {
+        private final BlockFactory blockFactory;
+        private final EvalOperator.ExpressionEvaluator field;
+        private final boolean order;
+        private final TriFunction<BlockFactory, Block, Boolean, Block> sort;
```

Comment:
I'm a little torn on if this should be an `abstract` function on `Evaluator`. It doesn't make a difference and I wouldn't change it, but sometimes that makes things easier to read. 

One thing that would be nice is if the `Evaluator`'s `toString` had the field type in it somehow. It'd make the `profile` output a bit easier more descriptive. Not a huge deal though.